lib/matchers/indexes/v3/have_index_for.rb

```diff
@@ -1,42 +1,39 @@
 module Mongoid
   module Matchers
-    class HaveIndexForMongoid3 # :nodoc:
-      def initialize(index_fields)
-        @index_fields = index_fields.symbolize_keys!
-      end
 
-      def with_options(options = { })
-        @options = options
-        self
-      end
+    def have_index_for(index_key)
+      HaveIndexFor.new(index_key)
+    end
+
+    class HaveIndexFor < Mongoid::Matchers::Base::HaveIndexFor
 
       def matches?(klass)
         @klass  = klass.is_a?(Class) ? klass : klass.class
         @errors = []
 
         if @klass.respond_to?(:index_options)
           # Mongoid 3
-          unless @klass.index_options[@index_fields]
-            @errors.push "no index for #{@index_fields}"
+          unless @klass.index_options[@index_key]
```

Comment:
I know this code was here before, but it's a good idea to use a property from the base class, so `index_key` instead of `@index_key` and `klass` instead of `@klass`. Properties should be declared like `attr_reader`.